sys/net/gnrc/network_layer/sixlowpan/gnrc_sixlowpan.c

```diff
@@ -222,8 +217,7 @@ static void _receive(gnrc_pktsnip_t *pkt)
     }
 #endif
     else {
-        DEBUG("6lo: dispatch %02" PRIx8 " ... is not supported\n",
-              dispatch[0]);
+        DEBUG("6lo: dispatch %02x... is not supported\n", dispatch[0]);
```

Comment:
`%x` expects type to be `unsigned`. Please cast `dispatch[0]` to `unsigned` to avoid bad things from happening

Reply:
Dispatch is already `uint8_t` so it can only be upcasted. Nothing bad's gonna happen here ;-).

Reply:
I'm aware that in practice `float` gets promoted to `double` and all integer types smaller than `int` get promoted to `int`. But are you sure that is C standard? (At least ´clang -Weverything` does not complain, so it might very well be.) If not, an `uint8_t` will be pushed on the argument stack and an `unsigned int` will be popped by `printf`.

I personally would do the cast even if this is officially specified, as it just looks strange if format specifier and argument are of different type

Reply:
Well, this is how we did it in other places as well and it did not lead to problems (see basically most link-layer address outputs) so I believe a cast would just add unnecessary clutter.

Reply:
If the promotion to an `int` is indeed part of the C standard your code is fine. If not, this is a bug - regardless of whether this has caused problems on the platforms and toolchains tested so far.

Reply:
This would be the first time I here of such a bug and would mean we had to change a lot of the output-related code...

Reply:
As I said, I don't know whether this is indeed a bug or totally valid. I'll check the standard tomorrow. The fact that clang doesn't complain indicates that this is fine. I'll still prefer to be sure before giving the ACK.

Reply:
According to the standard integer promotion takes places for integer type arguments of variadic function. As a result `dispatch[0]` of type `uint8_t` will be promoted to type `int`. This is, therefore, safe to do.